src/story_protocol_python_sdk/resources/Group.py

```diff
@@ -0,0 +1,652 @@
+# src/story_protocol_python_sdk/resources/Group.py
+
+from web3 import Web3
+
+from story_protocol_python_sdk.abi.GroupingModule.GroupingModule_client import GroupingModuleClient
+from story_protocol_python_sdk.abi.GroupingWorkflows.GroupingWorkflows_client import GroupingWorkflowsClient
+from story_protocol_python_sdk.abi.IPAssetRegistry.IPAssetRegistry_client import IPAssetRegistryClient
+from story_protocol_python_sdk.abi.CoreMetadataModule.CoreMetadataModule_client import CoreMetadataModuleClient
+from story_protocol_python_sdk.abi.LicensingModule.LicensingModule_client import LicensingModuleClient
+from story_protocol_python_sdk.abi.LicenseRegistry.LicenseRegistry_client import LicenseRegistryClient
+from story_protocol_python_sdk.abi.PILicenseTemplate.PILicenseTemplate_client import PILicenseTemplateClient
+from story_protocol_python_sdk.abi.IPAccountImpl.IPAccountImpl_client import IPAccountImplClient
+
+from story_protocol_python_sdk.utils.license_terms import LicenseTerms
+from story_protocol_python_sdk.utils.transaction_utils import build_and_send_transaction
+from story_protocol_python_sdk.utils.sign import Sign
+from story_protocol_python_sdk.utils.constants import ZERO_ADDRESS, ZERO_HASH
+
+
+class Group:
+    """
+    A class to manage groups on Story Protocol.
+
+    :param web3 Web3: An instance of Web3.
+    :param account: The account to use for transactions.
+    :param chain_id int: The ID of the blockchain network.
+    """
+    def __init__(self, web3: Web3, account, chain_id: int):
+        self.web3 = web3
+        self.account = account
+        self.chain_id = chain_id
+
+        self.grouping_module_client = GroupingModuleClient(web3)
+        self.grouping_workflows_client = GroupingWorkflowsClient(web3)
+        self.ip_asset_registry_client = IPAssetRegistryClient(web3)
+        self.core_metadata_module_client = CoreMetadataModuleClient(web3)
+        self.licensing_module_client = LicensingModuleClient(web3)
+        self.license_registry_client = LicenseRegistryClient(web3)
+        self.pi_license_template_client = PILicenseTemplateClient(web3)
+
+        self.license_terms_util = LicenseTerms(web3)
+        self.sign_util = Sign(web3, self.chain_id, self.account)
+
+    def register_group(self, group_pool: str, tx_options: dict = None) -> dict:
+        """
+        Registers a Group IPA.
+
+        :param group_pool str: The address of the group pool.
+        :param tx_options dict: [Optional] The transaction options.
+        :return dict: A dictionary with the transaction hash and group ID.
+        """
+        try:
+            if not Web3.is_address(group_pool):
```

Comment:
Sometimes you use `Web3.is_address` as in this case and lines 142, 145 and 507, but sometimes you use `self.web3.is_address` as in lines 240, 243, 370, 375, 435, 446 and 459. Is there any reason for this inconsistency?